components/object_store/src/obkv/mod.rs

```diff
@@ -0,0 +1,929 @@
+        max_upload_concurrency: usize,
+    ) -> Result<Self> {
+        let shard_manager = ShardManager::try_new(client.clone(), shard_num).map_err(|source| {
+            StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            }
+        })?;
+        let meta_manager: MetaManager<T> =
+            MetaManager::try_new(client.clone()).map_err(|source| StoreError::Generic {
+                store: OBKV,
+                source: Box::new(source),
+            })?;
+        let upload_id = AtomicI64::new(0);
+        Ok(Self {
+            shard_manager,
+            meta_manager: Arc::new(meta_manager),
+            client,
+            current_upload_id: upload_id,
+            part_size,
+            max_object_size,
+            max_upload_concurrency,
+        })
+    }
+
+    #[inline]
+    fn check_size(&self, bytes: &Bytes) -> std::result::Result<(), Error> {
+        ensure!(
+            bytes.len() < self.max_object_size,
+            TooLargeData {
+                size: bytes.len(),
+                limit: self.max_object_size,
+            }
+        );
+
+        Ok(())
+    }
+
+    #[inline]
+    fn normalize_path(location: Option<&Path>) -> Path {
+        if let Some(path) = location {
+            if !path.as_ref().ends_with(DELIMITER) {
+                return Path::from(format!("{}{}", path.as_ref(), DELIMITER));
+            }
+            path.clone()
+        } else {
+            Path::from("")
+        }
+    }
+
+    #[inline]
+    pub fn pick_shard_table(&self, path: &Path) -> &str {
+        self.shard_manager.pick_shard_table(path)
+    }
+}
+
+impl<T: TableKv> ObkvObjectStore<T> {
+    async fn read_meta(&self, location: &Path) -> std::result::Result<ObkvObjectMeta, Error> {
+        let meta = self
+            .meta_manager
+            .read_meta(location)
+            .await
+            .box_err()
+            .context(ReadMeta {
+                path: location.as_ref().to_string(),
+            })?;
+
+        if let Some(m) = meta {
+            Ok(m)
+        } else {
+            MetaNotExists {
+                path: location.as_ref().to_string(),
+            }
+            .fail()
+        }
+    }
+
+    async fn get_internal(&self, location: &Path) -> std::result::Result<GetResult, Error> {
+        let meta = self.read_meta(location).await?;
+        let table_name = self.pick_shard_table(location);
+        let mut futures = FuturesOrdered::new();
+        for path in meta.parts {
+            let client = self.client.clone();
+            let table_name = table_name.to_string();
+            let future = async move {
+                match client.get(&table_name, path.as_bytes()) {
+                    Ok(res) => Ok(Bytes::from(res.unwrap())),
+                    Err(err) => Err(StoreError::Generic {
+                        store: OBKV,
+                        source: Box::new(err),
+                    }),
+                }
```

Comment:
I guess we can use `get_batch` to avoid extra IO problems. And Maybe we can leave a TODO for now.

Reply:
Ok, I leave a a TODO here.